src/mito/src/engine.rs

```diff
@@ -502,9 +519,16 @@ impl<S: StorageEngine> MitoEngineInner<S> {
 
         logging::info!("start altering table {} with request {:?}", table_name, req);
         table
-            .alter(AlterContext::new(), req)
+            .alter(AlterContext::new(), &req)
             .await
             .context(error::AlterTableSnafu { table_name })?;
+
+        if let AlterKind::RenameTable { new_table_name } = &req.alter_kind {
+            table_ref.table = new_table_name.as_str();
+            let mut tables = self.tables.write().unwrap();
+            tables.remove(&table_ref.to_string());
+            tables.insert(table_ref.to_string(), table.clone());
```

Comment:
We could allocate the full name first.
```suggestion
            table_ref.table = new_table_name.as_str();
            let full_table_name = table_ref.to_string();
            let mut tables = self.tables.write().unwrap();
            tables.remove(&full_table_name);
            tables.insert(full_table_name, table.clone());
```